File: src/main/java/com/moovie/backend/domain/model/MovieDetail.java
package com.moovie.backend.domain.model;

import java.util.List;

public record MovieDetail(
        int id,
        String title,
        String overview,
        String posterPath,
        String backdropPath,
        double voteAverage,
        String releaseDate,
        String tagline,
        Integer runtime,
        List<String> genres,
        String director,
        List<String> cast,
        List<WatchProvider> watchProviders,
        List<Movie> similarMovies,
        List<MovieReview> popularReviews,
        int reviewCount,
        int listCount,
        int likeCount
) {
}

File: src/main/kotlin/org/example/domain/model/MovieDetail.kt
package org.example.domain.model

import kotlinx.serialization.Serializable

@Serializable
data class MovieDetail(
    val id: Int,
    val title: String,
    val overview: String,
    val posterPath: String? = null,
    val backdropPath: String? = null,
    val voteAverage: Double = 0.0,
    val releaseDate: String? = null,
    val tagline: String? = null,
    val runtime: Int? = null,
    val genres: List<String> = emptyList(),
    val director: String? = null,
    val cast: List<String> = emptyList(),
    val watchProviders: List<WatchProvider> = emptyList(),
    val similarMovies: List<Movie> = emptyList(),
    val popularReviews: List<MovieReview> = emptyList(),
    val reviewCount: Int = 0,
    val listCount: Int = 0,
    val likeCount: Int = 0
)
